core/rfb.js

```diff
@@ -688,6 +688,12 @@ export default class RFB extends EventTargetMixin {
         }
 
         const size = this._screenSize();
+
+        // Don't request the same size we already have
+        if (this._fbWidth === size.w && this._fbHeight === size.h) {
+            return;
+        }
```

Comment:
This could in theory be race:y, but I'm not sure it will ever happen in practice.

If you've sent a request but haven't gotten a reply yet, then the values you are checking are stale.

E.g.

1. Start at 100×100
2. Send request to 200×200
3. Before you've gotten a reply, try to resize back to 100×100

This scenario will now break since no request will be sent in 3. The session will be stuck at 200×200 even though 100×100 was the latest proper size.

Reply:
Good point, I did not consider this. We could save the previous request values and check against those I guess, but I don't think it's worth the added complexity.

I'll remove this change, the gain from it is minimal anyway.